tests/ModelTest.php

```diff
@@ -924,6 +920,59 @@ public function testRaw(): void
         $this->assertNotNull($result);
     }
 
+    #[DataProvider('provideTypeMap')]
+    public function testRawHyradeModel(array $typeMap): void
+    {
+        User::insert([
+            ['name' => 'John Doe', 'age' => 35, 'embed' => ['foo' => 'bar'], 'list' => [1, 2, 3]],
+            ['name' => 'Jane Doe', 'age' => 35, 'embed' => ['foo' => 'bar'], 'list' => [1, 2, 3]],
+            ['name' => 'Harry Hoe', 'age' => 15, 'embed' => ['foo' => 'bar'], 'list' => [1, 2, 3]],
+        ]);
+
+        // Single document result
+        $user = User::raw(fn (Collection $collection) => $collection->findOne(
+            ['age' => 35],
+            [
+                'projection' => ['_id' => 1, 'name' => 1, 'age' => 1, 'now' => '$$NOW', 'embed' => 1, 'list' => 1],
+                'typeMap' => $typeMap,
+            ],
+        ));
+
+        $this->assertInstanceOf(User::class, $user);
+        $this->assertArrayNotHasKey('_id', $user->getAttributes());
+        $this->assertArrayHasKey('id', $user->getAttributes());
+        $this->assertNotEmpty($user->id);
+        $this->assertInstanceOf(Carbon::class, $user->now);
+        $this->assertEquals(['foo' => 'bar'], (array) $user->embed);
+        $this->assertEquals([1, 2, 3], (array) $user->list);
+
+        // Cursor result
+        $result = User::raw(fn (Collection $collection) => $collection->aggregate([
+            ['$set' => ['now' => '$$NOW']],
+            ['$limit' => 2],
+        ], ['typeMap' => $typeMap]));
+
+        $this->assertInstanceOf(EloquentCollection::class, $result);
+        $this->assertCount(2, $result);
+        $user = $result->first();
+        $this->assertInstanceOf(User::class, $user);
+        $this->assertArrayNotHasKey('_id', $user->getAttributes());
+        $this->assertArrayHasKey('id', $user->getAttributes());
+        $this->assertNotEmpty($user->id);
+        $this->assertInstanceOf(Carbon::class, $user->now);
+        $this->assertEquals(['foo' => 'bar'], $user->embed);
+        $this->assertEquals([1, 2, 3], $user->list);
+    }
+
+    public static function provideTypeMap(): Generator
+    {
+        yield 'default' => [[]];
+        yield 'array' => [['root' => 'array', 'document' => 'array', 'array' => 'array']];
+        yield 'object' => [['root' => 'object', 'document' => 'object', 'array' => 'array']];
+        yield 'Library BSON' => [['root' => BSONDocument::class, 'document' => BSONDocument::class, 'array' => BSONArray::class]];
+        yield 'Driver BSON' => [['root' => 'bson', 'document' => 'bson', 'array' => 'bson']];
```

Comment:
Nit: when `root` is `bson`, the `document` and `array` types are ignored completely.
```suggestion
        yield 'Driver BSON' => [['root' => 'bson']];
```